src/main/webapp/search.html

```diff
@@ -4,11 +4,15 @@
     <meta charset="UTF-8">
     <title>Search Page</title>
     <link rel="stylesheet" href="style.css">
-    <script src="script.js"></script>
+    <script src="loginScript.js"></script>
```

Comment:
With the simple site you're creating here, it might be easier to have a dedicated login.html, then a logout button that redirects to that page after signaling the server to log out.

Then if a user goes to a page but isn't logged in, it will redirect them to that page first.

Otherwise you'll have to put this on every page, and handle some weird edge cases where they click between pages but the server response hasn't come back yet/etc causing user frustration as they have to login yet again.

Reply:
As with most of my comments here, this could be done in a future PR, but it's something to keep in mind.